test/integration/ingress_webhook_test.go

```diff
@@ -0,0 +1,362 @@
+				ObjectMeta: metav1.ObjectMeta{
+					Name: uuid.NewString(),
+				},
+				Spec: netv1.IngressSpec{
+					IngressClassName: lo.ToPtr(consts.IngressClass),
+					Rules: []netv1.IngressRule{
+						{
+							IngressRuleValue: netv1.IngressRuleValue{
+								HTTP: &netv1.HTTPIngressRuleValue{
+									Paths: []netv1.HTTPIngressPath{
+										constructIngressPathImplSpecific("/bar"),
+										constructIngressPathImplSpecific("/~baz[1-9]"),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			WantCreateErr: false,
+		},
+		testCaseIngressValidation{
+			Name: "invalid regex path fails validation",
+			Ingress: &netv1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: uuid.NewString(),
+				},
+				Spec: netv1.IngressSpec{
+					IngressClassName: lo.ToPtr(consts.IngressClass),
+					Rules: []netv1.IngressRule{
+						{
+							IngressRuleValue: netv1.IngressRuleValue{
+								HTTP: &netv1.HTTPIngressRuleValue{
+									Paths: []netv1.HTTPIngressPath{
+										constructIngressPathImplSpecific("/bar"),
+										constructIngressPathImplSpecific("/~baz[1-9]"),
+									},
+								},
+							},
+						},
+						{
+							IngressRuleValue: netv1.IngressRuleValue{
+								HTTP: &netv1.HTTPIngressRuleValue{
+									Paths: []netv1.HTTPIngressPath{
+										constructIngressPathImplSpecific("/~foo[[["),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			WantCreateErr:          true,
+			WantCreateErrSubstring: "regex parse error:\n    ^foo[[[\n          ^\nerror: unclosed character class",
+		},
+	)
+	testIngressValidationWebhook(ctx, t, namespace, testCases)
+}
+
+// setUpEnvForTestingIngressValidationWebhook sets up the environment for testing Ingress validation webhook,
+// it sets it only for objects applied to namespace specified as argument.
+func setUpEnvForTestingIngressValidationWebhook(ctx context.Context, t *testing.T) (namespace string) {
+	ns, _ := helpers.Setup(ctx, t, env)
+	namespace = ns.Name
+	const webhookName = "kong-validations-ingress"
+	ensureAdmissionRegistration(
+		ctx,
+		t,
+		namespace,
+		webhookName,
+		[]admregv1.RuleWithOperations{
+			{
+				Rule: admregv1.Rule{
+					APIGroups:   []string{"networking.k8s.io"},
+					APIVersions: []string{"v1"},
+					Resources:   []string{"ingresses"},
+				},
+				Operations: []admregv1.OperationType{admregv1.Create, admregv1.Update},
+			},
+		},
+	)
+	ensureWebhookServiceIsConnective(ctx, t, webhookName)
+	return namespace
+}
+
+// testIngressValidationWebhook tries to create the given Ingress (passed in testCaseIngressValidation) and asserts expected results.
+func testIngressValidationWebhook(
+	ctx context.Context, t *testing.T, namespace string, testCases []testCaseIngressValidation,
+) {
+	for _, tC := range testCases {
+		t.Run(tC.Name, func(t *testing.T) {
+			_, err := env.Cluster().Client().NetworkingV1().Ingresses(namespace).Create(ctx, tC.Ingress, metav1.CreateOptions{})
+			if tC.WantCreateErr {
```

Comment:
Nit: we could use just `WantCreateErrSubstring`:

```suggestion
			if tC.WantCreateErrSubstring != "" {
```